### PR TITLE
Added HTTP Authentication

### PR DESCRIPTION
**Pull request by akifrabbani**
Added support for simple HTTP based authentication.

`{
	authentication: 'HTTPAuth',
	authenticationOptions: {
		url: 'https://my-website.com/api/backend-login'
	}
}`

It will send a JSON POST request to specified URL and expecting status code 200 in order to be passed as successful authentication.

### CODE REVIEW

File: src/auth/HTTPAuth.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+export class HTTPAuth implements IAuthentication {
+	private url: string;
+
+	constructor(config: IHTTPAuthOptions) {
+		this.url = config.url;
+	}
+
+	async authenticate(username: string, password: string) {
+
```
Comment (simllll) [inline src/auth/HTTPAuth.ts:16]: please remove empty lines, ensure you have linted the file with eslint :-)

File: src/auth/HTTPAuth.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+		let success = false;
+		try {
+			
+			await axios
+			  .post(this.url, {
+			    username: username,
```
Comment (simllll) [inline src/auth/HTTPAuth.ts:22]: use shorthand (eslint fix should do that for you)

File: src/auth/HTTPAuth.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+			    password: password
+			  })
```
Comment (simllll) [inline src/auth/HTTPAuth.ts:24]: i would prefer "three" cases here:
1.) auth success
2.) auth failed
3.) other http error (e.g. server error like 5xx)

remove the try / catch block and just use
await axios..
but add the axios config:
```
validateStatus: function (status) {
  return status >= 200 && status < 500;
}
```

afterwards all 200 till 499 codes are "resolved" and will not throw, but 5xx will throw (as exepcted)


e.g. somethign like this (pseudo code):

```ts
const res = await axios
			  .post(this.url, {
			    username,
			    password
			  }, validateStatus: function (status) {
  return status >= 200 && status < 500;
});

if (res.status == 200) { 
  /// Return code 200 -> authentication successful.
  return true;
  }

return false;

```

**Review by simllll — COMMENTED** [review]
thanks for your contribution, I just reviewed it :-)

**Comment by akifrabbani** [comment]
Added validateStatus and eslinted.

**Comment by simllll** [comment]
thanks, I will simplify the code after I merged it and remove the try / catch part.

thanks for your contribution!

**Comment by simllll** [comment]
Released in 1.2.0

**Pull request merged**